src/sentry/static/sentry/app/components/eventOrGroupHeader.jsx

```diff
@@ -56,7 +57,7 @@ class EventOrGroupHeader extends React.Component {
     if (includeLink) {
       props.to = `/${orgId}/${projectId}/issues/${isEvent ? groupID : id}/${isEvent
         ? `events/${data.id}/`
-        : ''}`;
+        : ''}${this.props.query ? `?query=${this.props.query}` : ''}`;
```

Comment:
@billyvg is this the right way to do this? Feels odd.

Reply:
Technically it works - it's very difficult to read though

You could try to define `to` as an object like this: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md

Reply:
Does this look right? https://github.com/getsentry/sentry/pull/7538/commits/3898a2e8d36623851cdc389e6f59c485872fae3c

Reply:
Yeah looks right